source/AI.cpp

```diff
@@ -2573,6 +2630,107 @@ bool AI::DoCloak(Ship &ship, Command &command)
 
 
 
+bool AI::DoRoving(Ship &ship, Command &command, unsigned int rovingShipCount)
+{
+	const System* system = ship.GetSystem();
+	if((rovingShipCount > 9 && system))
+	{
+		if(ship.Fuel() >= ship.JumpFuel() || ship.Fuel() >= ship.JumpDriveFuel())
+		{
+			MoveIndependent(ship, command);
+			return true;
+		}
+		else if (system->IsInhabited(&ship))
+		{
+			for(const StellarObject& planet: system->Objects())
+				if(CanRefuel(ship, &planet))
+				{
+					ship.SetTargetStellar(&planet);
+				}
+			Refuel(ship, command);
+			return true;
+		}
+	}
+	
+	const Point target = ship.GetTargetPosition();
+	const auto v = ship.MaxVelocity();
+    if(!target || MoveTo(ship, command, target, Point(), v, v))
+    {
+        Point newTarget = Angle::Random().Unit() * Random::Real() * MAX_DISTANCE_FROM_CENTER;
+        ship.SetTargetPosition(newTarget);
+    }
+    
+    return false;
+}
+
+
+
+bool AI::DoEvasive(Ship &ship, Command &command)
+{
+	// Did I have to move my targetPosition?
+	bool movedTarget = false;
+	
+	// First, figure out which ships are not of the same government. Ignore disabled/destroyed ships.
+	vector<shared_ptr<Ship>> enemies;
+	for(const auto &otherShip : GetShipsList(ship, false))
```

Comment:
Would it be useful to limit the max range where enemies can be that are considered?

Reply:
Probably not, because it's dependent on enemy weapons range, not just proximity.